wordpress-mobile/WordPress-iOS
# Issue: Reset the tabbar selectedIndex when resetting the tabbar/splitViewControllers.

Fixes the issue mentioned in https://github.com/wordpress-mobile/WordPress-iOS/pull/6451#issuecomment-274785738

What was happening here was that once a user logs out and then back in, the `WPTabBarController` receives a notification and calls `reloadSplitViewControllers` to reset. However, when logging back in during the same session, the previously `selectedIndex` was still set for the Me tab which then loads and displays after reset. This differed from the load order of a new user logging in where the My Sites tab is initially selected and loads `BlogDetailsViewController` for us.

It is within `BlogDetailsViewController` that, when loaded, calls `syncBlogAndAllMetadata` for the corresponding blog, which will give us the categories and post formats for when the user opens up the Editor Options.

So now we just reset the index to always load the My Sites tab.

To test:
1. Be signed in to a .COM account.
2. Open Me tab, "Disconnect".
3. Keep the app running, log back in.
4. The My Sites tab should be selected.
5. Open the Editor > Options.
6. Categories and Post Formats should be available.

Needs review: @frosty can you check this out?
